transformers/modeling_unilm.py

```diff
@@ -0,0 +1,747 @@
+# coding=utf-8
+# The MIT License (MIT)
+
+# Copyright (c) Microsoft Corporation
+
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+"""PyTorch UniLM model. """
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+import sys
+import copy
+import math
+import logging
+import tarfile
+import tempfile
+import shutil
+import numpy as np
+
+import torch
+from torch import nn
+from torch.nn import CrossEntropyLoss, MSELoss
+import torch.nn.functional as F
+from torch.nn.modules.loss import _Loss
+
+from .file_utils import cached_path
+from .modeling_utils import PreTrainedModel
+from .configuration_unilm import UnilmConfig
+from .modeling_bert import ACT2FN, load_tf_weights_in_bert, BertPooler, BertIntermediate, BertOutput, BertPredictionHeadTransform, BertSelfOutput, BertLMPredictionHead, BertOnlyMLMHead, BertOnlyMLMHead, BertEmbeddings
+
+logger = logging.getLogger(__name__)
+
+UNILM_PRETRAINED_MODEL_ARCHIVE_MAP = {
```

Comment:
The current checkpoints are hosted at a temporary place. How could we upload them to the huggingface aws storage? @rlouf

Reply:
Hi @donglixp, you can create an account at https://huggingface.co/join then use the cli from inside the library `transformers-cli login` then `transformers-cli upload`.

We are in the process of improving the documentation on this subject today, cc @LysandreJik @thomwolf

Reply:
> Hi @donglixp, you can create an account at https://huggingface.co/join then use the cli from inside the library `transformers-cli login` then `transformers-cli upload`.
> 
> We are in the process of improving the documentation on this subject today, cc @LysandreJik @thomwolf

Thanks for the pointer!